python-lib/ontology_tagger.py

```diff
@@ -108,14 +119,24 @@ def _tokenize_keywords(
             List: The tokenized keywords.
 
         """
-        keywords = [get_keyword(keyword, self.normalize_case) for keyword in keywords]
+        if self.lemmatization:
+            self.tokenizer._activate_components_to_lemmatize(language)
+        keywords = [
+            normalize_case(keyword, self.normalize_case) for keyword in keywords
+        ]
         tokenized_keywords = list(
-            self.tokenizer.spacy_nlp_dict[language].tokenizer.pipe(keywords)
+            self.tokenizer.spacy_nlp_dict[language].pipe(keywords)
         )
-        self._keyword_to_tag[language] = {
-            get_keyword(keyword.text, self.normalize_case): tag
-            for keyword, tag in zip(tokenized_keywords, tags)
-        }
+        if self.lemmatization:
+            self._keyword_to_tag[language] = {
+                lemmatize(keyword, self.normalize_case): tag
```

Comment:
Better to either use a big function called sthg like "normalize" which has a lemmatize and normalize_case argument, or two smaller functions.

Reply:
has been solved here d807407